### ISSUE TITLE
[Patch] stack trace when accessing a FQDN that does not resolve is not helping

### ISSUE THREAD

**Issue by tpo** [qgiscloud/qgis-cloud-plugin#51]
When I enter a wrong server FQDN in the "Server" field in the qgis-cloud-plugin dialog I get this stack trace:

```
Ein Fehler trat bei der Ausführung von Python-Code auf: 

urllib.error.URLError: 
Traceback (most recent call last):
  File "/usr/lib/python3.7/urllib/request.py", line 1324, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/usr/lib/python3.7/http/client.py", line 1244, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1290, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1239, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/usr/lib/python3.7/http/client.py", line 966, in send
    self.connect()
  File "/usr/lib/python3.7/http/client.py", line 1399, in connect
    super().connect()
  File "/usr/lib/python3.7/http/client.py", line 938, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib/python3.7/socket.py", line 707, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/lib/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Der Name oder der Dienst ist nicht bekannt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudplugindialog.py", line 295, in check_login
    version_info=self._version_info())
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudapi/qgiscloudapi.py", line 161, in check_login
    content = request.post(resource, data)
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudapi/qgiscloudapi.py", line 596, in post
    return self.request(resource, method='POST', data=data)
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudapi/qgiscloudapi.py", line 684, in request
    response = urllib.request.urlopen(req).read()
  File "/usr/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/usr/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.7/urllib/request.py", line 1367, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib/python3.7/urllib/request.py", line 1326, in do_open
    raise URLError(err)
urllib.error.URLError: 


Python-Version: 3.7.3 (default, Jul 25 2020, 13:03:44) [GCC 8.3.0] 
QGIS-Version: 3.16.4-Hannover Hannover, 654e76b3cb 

Python-Pfad:
/usr/share/qgis/python
/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python
/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins
/usr/share/qgis/python/plugins
/usr/lib/python37.zip
/usr/lib/python3.7
/usr/lib/python3.7/lib-dynload
/home/tpo/.local/lib/python3.7/site-packages
/usr/local/lib/python3.7/dist-packages
/usr/lib/python3/dist-packages
/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python
```

**Comment by tpo** [comment]
I suggest to improve the stacktrace with this patch:

```
--- qgiscloudapi.py.old 2021-03-08 17:42:01.298130189 +0100
+++ qgiscloudapi.py     2021-03-08 17:39:09.631641598 +0100
@@ -44,7 +44,7 @@
 API_URL = 'https://api.qgiscloud.com'
 
 __all__ = ['API', 'UnauthorizedError', 'ConnectionException', 'TokenRequiredError', 'BadRequestError', 'ForbiddenError',
-           'ConflictDuplicateError', 'GoneError', 'InternalServerError',
+           'ConflictDuplicateError', 'GoneError', 'InternalServerError', 'URLError',
            'NotImplementedError', 'ThrottledError']
 
 
@@ -461,6 +461,20 @@
         msg = '\n'.join(self.msgs)
         return msg
 
+class URLError(Exception):
+    """
+        We raise this exception whenever we get an URLError from urllib,
+        one reason of which could be an invalid FQDN/host name.
+    """
+
+    msg = ""
+
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __str__(self):
+        return self.msg
+
 class UnauthorizedError(Exception):
     """
         We raise this exception whenever the API answers with HTTP STATUS 401
@@ -683,6 +697,8 @@
                 req.get_method = lambda: request_method
             response = urllib.request.urlopen(req).read()
 
+        except urllib.error.URLError as e:
+            raise URLError("Failed to access '{}'. '{}'".format(url, e.reason))
         except urllib.error.HTTPError as e:
             #
             # Handle the possible responses according to their HTTP STATUS
```

That will change the strack trace into the following which actually spells out what's going wrong:

```
Ein Fehler trat bei der Ausführung von Python-Code auf: 

qgiscloud.qgiscloudapi.qgiscloudapi.URLError: Failed to access 'https://foobar.qgiscloud.com/notifications.json'. '[Errno -2] Der Name oder der Dienst ist nicht bekannt' 
Traceback (most recent call last):
  File "/usr/lib/python3.7/urllib/request.py", line 1324, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/usr/lib/python3.7/http/client.py", line 1244, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1290, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1239, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/usr/lib/python3.7/http/client.py", line 966, in send
    self.connect()
  File "/usr/lib/python3.7/http/client.py", line 1399, in connect
    super().connect()
  File "/usr/lib/python3.7/http/client.py", line 938, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib/python3.7/socket.py", line 707, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/lib/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Der Name oder der Dienst ist nicht bekannt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudapi/qgiscloudapi.py", line 698, in request
    response = urllib.request.urlopen(req).read()
  File "/usr/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/usr/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.7/urllib/request.py", line 1367, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib/python3.7/urllib/request.py", line 1326, in do_open
    raise URLError(err)
urllib.error.URLError: 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudplugindialog.py", line 295, in check_login
    version_info=self._version_info())
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudapi/qgiscloudapi.py", line 161, in check_login
    content = request.post(resource, data)
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudapi/qgiscloudapi.py", line 610, in post
    return self.request(resource, method='POST', data=data)
  File "/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins/qgiscloud/qgiscloudapi/qgiscloudapi.py", line 701, in request
    raise URLError("Failed to access '{}'. '{}'".format(url, e.reason))
qgiscloud.qgiscloudapi.qgiscloudapi.URLError: Failed to access 'https://foobar.qgiscloud.com/notifications.json'. '[Errno -2] Der Name oder der Dienst ist nicht bekannt'


Python-Version: 3.7.3 (default, Jul 25 2020, 13:03:44) [GCC 8.3.0] 
QGIS-Version: 3.16.4-Hannover Hannover, 654e76b3cb 

Python-Pfad:
/usr/share/qgis/python
/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python
/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python/plugins
/usr/share/qgis/python/plugins
/usr/lib/python37.zip
/usr/lib/python3.7
/usr/lib/python3.7/lib-dynload
/home/tpo/.local/lib/python3.7/site-packages
/usr/local/lib/python3.7/dist-packages
/usr/lib/python3/dist-packages
/home/tpo/.local/share/QGIS/QGIS3/profiles/default/python
```

**Comment by hdus** [comment]
Please create a pull request.

**Comment by tpo** [comment]
Implemented via 461c0d7
